sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -59,6 +61,141 @@ class BucketedReadSuite extends QueryTest with SQLTestUtils with TestHiveSinglet
     }
   }
 
+  // To verify if pruning works, we compare the results before filtering
+  private def checkPrunedAnswers(
+      sourceDataFrame: DataFrame,
+      filterCondition: Column,
```

Comment:
instead of having these 2 parameters, how about we just ask caller to pass in a `filteredDataFrame`?